pulsar-client-cpp/lib/ProducerConfiguration.cc

```diff
@@ -73,6 +73,18 @@ ProducerConfiguration& ProducerConfiguration::setMaxPendingMessages(int maxPendi
 
 int ProducerConfiguration::getMaxPendingMessages() const { return impl_->maxPendingMessages; }
 
+ProducerConfiguration& ProducerConfiguration::setMaxPendingMessagesAcrossPartitions(int maxPendingMessages) {
+    if (maxPendingMessages <= 0) {
```

Comment:
Assert instead of throw

Reply:
Or leave it as an assumption since you are checking argument in ProducerConfiguration